### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by j-tee**
## In this milestone:
- Use [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [`axios`](https://axios-http.com/docs/intro) with [`createAsyncThunk`](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.
- Fetch book list from Bookstore API on Books page load.
    - Add an Action Creator, that returns a function.
    - Fetch list of books.
    - Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- Refactor *add book* and *remove book* features to persist changes in the server.
  - Change existing Action Creators, to return functions. 
  - Make changes by using API.
  - Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- *Add book* and *remove book* work in the same way after the refactor.

### CODE REVIEW

File: src/redux/book/books.js [hunk 1]
```diff
@@ -1,36 +1,27 @@
+/* eslint-disable no-console */
 /* eslint-disable no-undef */
+// import { objectTraps } from 'immer/dist/internal';
```
Comment (rustam-tolipov) [inline src/redux/book/books.js]: You did great job, however, there is a a tiny issue.
To make your code professional, kindly remove `commented` codes from your file 😊

File: src/components/Buttons.js [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+  const {
+    id, className, handleClickEvent, name, type,
+  } = props;
+  return (
+    <button
+      id={id}
+      className={className}
+      onClick={handleClickEvent}
+      name={name}
+      type={type}
```
Comment (rustam-tolipov) [inline src/components/Buttons.js:14]: [OPTIONAL] You don't have to make props for type of your button. Otherwise linters may show an error for it.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @j-tee, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Kindly add MIT license to your project 🪪 👮🏻‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED ✔️ 🟢 

## Hi @j-tee 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**